middleware/src/main/java/br/ufrn/dimap/middleware/MiddlewareConfig.java

```diff
@@ -0,0 +1,163 @@
+/**
+ * 
+ */
+package br.ufrn.dimap.middleware;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import br.ufrn.dimap.middleware.extension.interfaces.ClientProtocolPlugIn;
+import br.ufrn.dimap.middleware.extension.interfaces.InvocationInterceptorSerialized;
+import br.ufrn.dimap.middleware.extension.interfaces.InvocationInterceptorUnserialized;
+import br.ufrn.dimap.middleware.extension.interfaces.ServerProtocolPlugin;
+
+/**
+ * This static class serves as a facade for the configuration of the
+ * middleware. Interceptors, plugins and other features are registered
+ * here and are available for the whole  
+ * 
+ * @author Gustavo Carvalho
+ *
+ */
+public class MiddlewareConfig {
+	
+	/**
+	 * This nested class holds the information regarding the interceptors of the framework
+	 * 
+	 * @author Gustavo Carvalho
+	 */
+	public static class Interceptors {
```

Comment:
Consider changing this to a Singleton, instead of using static for all methods and attributes.